freshbooks/api/business.py

```diff
@@ -0,0 +1,181 @@
+
+from freshbooks.api.resource import HttpVerbs, Resource
+from freshbooks.builders import Builder
+from freshbooks.builders.includes import IncludesBuilder
+from freshbooks.errors import FreshBooksError, FreshBooksNotImplementedError
+from freshbooks.models import ListResult, Result, VisState
+
+
+class BusinessResource(Resource):
+    """Handles resources under the `/accounting` endpoints."""
+
+    def __init__(self, client_config: SimpleNamespace, accounting_path: str,
+                 single_name: str, list_name: str,
+                 delete_via_update: bool = True, missing_endpoints: Optional[List[str]] = None):
+        super().__init__(client_config)
+        self.accounting_path = accounting_path
+        self.single_name = single_name
+        self.list_name = list_name
+        self.delete_via_update = delete_via_update
+        self.missing_endpoints = missing_endpoints or []
+
+    def _get_url(self, account_id: str, resource_id: Optional[int] = None) -> str:
+        if resource_id:
+            return "{}/accounting/businesses/{}/{}/{}".format(
+                self.base_url, account_id, self.accounting_path, resource_id)
+        return "{}/accounting/businesses/{}/{}".format(self.base_url, account_id, self.accounting_path)
+
+    def _extract_error(self, errors: Union[list, dict]) -> Tuple[str, Optional[int]]:
+        if not errors:  # pragma: no cover
+            return "Unknown error", None
+
+        if isinstance(errors, list):
+            return errors[0]["message"], int(errors[0]["errno"])
+
+        return errors["message"], int(errors["errno"])  # pragma: no cover
+
+    def _request(self, url: str, method: str, data: Optional[dict] = None) -> Any:
+        response = self._send_request(url, method, data)
+
+        status = response.status_code
+        if status == 200 and method == HttpVerbs.HEAD:  # pragma: no cover
+            # no content returned from a HEAD
+            return
+
+        try:
+            content = response.json(parse_float=Decimal)
+        except ValueError:
+            raise FreshBooksError(status, "Failed to parse response", raw_response=response.text)
+
+        if "data" not in content:
+            raise FreshBooksError(status, "Returned an unexpected response", raw_response=response.text)
+
+        response_data = content
+        if status >= 400:
+            message, code = self._extract_error(response_data.get("errors")) if response_data.get("errors") else ("unknown error", status)
+            raise FreshBooksError(status, message, error_code=code, raw_response=content)
+        try:
+            return response_data["result"]
+        except KeyError:
+            return response_data
+
+    def _reject_missing(self, name: str) -> None:
+        if name in self.missing_endpoints:
+            raise FreshBooksNotImplementedError(self.list_name, name)
+
+    def get(self, account_id: str, resource_id: int, includes: Optional[IncludesBuilder] = None) -> Result:
```

Comment:
`resource_id: int` is incorrect on these endpoints as they take a uuid now instead of an int. `resource_id: str` instead.

Reply:
Fixed